exporters/variables-scss/src/generators/cssObjectGenerator.ts

```diff
@@ -1,51 +1,56 @@
-import { Token, TokenGroup } from '@supernovaio/sdk-exporters';
-import { toPlural } from '../helpers/stringHelper';
+import { Token, TokenGroup, TokenType } from '@supernovaio/sdk-exporters';
 import { tokenVariableName } from '../helpers/tokenHelper';
+import { toPlural } from '../helpers/stringHelper';
 
-export const generateObjectContent = (
-  tokens: Array<Token>,
-  tokenGroups: Array<TokenGroup>,
-  withParent: boolean,
-): string => {
-  return tokens.reduce((result, token) => {
-    const name = tokenVariableName(token, tokenGroups, withParent);
-    const numericPart = name.match(/\d+/)?.[0];
-    const prefix = `${token.origin?.name?.split('/')[0].toLowerCase()}-`;
-    const nonNumericPart = name.replace(prefix, '');
-
-    if (numericPart) {
-      result += `${numericPart}: $${name},\n`;
-    } else if (nonNumericPart) {
-      result += `${nonNumericPart}: $${name},\n`;
-    }
+export type CssObjectType = { [key: string]: string | object };
+
+const specialCases: { [key: string]: string } = {
+  'radius-full': 'full',
+};
+
+export const getNonNumericPart = (tokenName: string): string => {
+  if (specialCases[tokenName]) {
+    return specialCases[tokenName];
+  }
 
-    return result;
-  }, '');
+  return tokenName.toLowerCase();
 };
 
 export const generateCssObjectFromTokens = (
   tokens: Array<Token>,
   mappedTokens: Map<string, Token>,
   tokenGroups: Array<TokenGroup>,
   hasParentPrefix: boolean,
-): string => {
-  const originNameMap = new Map<string, Array<Token>>();
-  tokens.forEach((token) => {
-    const originName = token.origin?.name;
-
-    if (originName) {
-      const nameParts = originName.split('/');
-      nameParts.pop();
-      const objectName = toPlural(nameParts.join('-').toLowerCase());
-      originNameMap.set(objectName, [...(originNameMap.get(objectName) || []), token]);
-    }
-  });
+): CssObjectType => {
+  return tokens.reduce((cssObject, token) => {
+    const nameParts = token.origin?.name?.split('/');
 
-  return Array.from(originNameMap.entries())
-    .map(([objectName, token]) => {
-      const objectContent = generateObjectContent(token, tokenGroups, hasParentPrefix);
+    if (nameParts) {
+      let currentObject: CssObjectType = cssObject;
```

Comment:
I am not sure if you are working with the reducer correctly. I think this code magically works because of this reference to this `currentObject`. But I think a better way should be fabricating a new `const currentObject` and adding it to the accumulator as a latest step, e.g. `return { ...cssObject, ...currentObject }`.

It also looks like you are intentionally or not accessing and existing parts of the accumulator `currentObject[part] = currentObject[part] || {};`.

Reply:
I don't know how to solve this without an object reference. I added a TODO